File: openapi-generator/csharp-dotnet2/src/main/CsharpDotNet2/Org/OpenAPITools/Model/TagModify.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace Org.OpenAPITools.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class TagModify {
    /// <summary>
    /// Gets or Sets Add
    /// </summary>
    [DataMember(Name="add", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "add")]
    public TagModifyAdd Add { get; set; }

    /// <summary>
    /// Gets or Sets Delete
    /// </summary>
    [DataMember(Name="delete", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "delete")]
    public TagModifyDelete Delete { get; set; }

    /// <summary>
    /// Updated name of this tag.
    /// </summary>
    /// <value>Updated name of this tag.</value>
    [DataMember(Name="name", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "name")]
    public string Name { get; set; }

    /// <summary>
    /// If this tag is part a hierarchical tag tree as a child tag, the parentTagId is the ID of this tag's parent tag.
    /// </summary>
    /// <value>If this tag is part a hierarchical tag tree as a child tag, the parentTagId is the ID of this tag's parent tag.</value>
    [DataMember(Name="parentTagId", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "parentTagId")]
    public long? ParentTagId { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class TagModify {\n");
      sb.Append("  Add: ").Append(Add).Append("\n");
      sb.Append("  Delete: ").Append(Delete).Append("\n");
      sb.Append("  Name: ").Append(Name).Append("\n");
      sb.Append("  ParentTagId: ").Append(ParentTagId).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: openapi-generator/dart/lib/model/tag_modify.dart
part of openapi.api;

class TagModify {
  
  TagModifyAdd add = null;
  
  TagModifyDelete delete = null;
  /* Updated name of this tag. */
  String name = null;
  /* If this tag is part a hierarchical tag tree as a child tag, the parentTagId is the ID of this tag's parent tag. */
  int parentTagId = null;
  TagModify();

  @override
  String toString() {
    return 'TagModify[add=$add, delete=$delete, name=$name, parentTagId=$parentTagId, ]';
  }

  TagModify.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['add'] == null) {
      add = null;
    } else {
      add = new TagModifyAdd.fromJson(json['add']);
    }
    if (json['delete'] == null) {
      delete = null;
    } else {
      delete = new TagModifyDelete.fromJson(json['delete']);
    }
    if (json['name'] == null) {
      name = null;
    } else {
          name = json['name'];
    }
    if (json['parentTagId'] == null) {
      parentTagId = null;
    } else {
          parentTagId = json['parentTagId'];
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'add': add,
      'delete': delete,
      'name': name,
      'parentTagId': parentTagId
    };
  }

  static List<TagModify> listFromJson(List<dynamic> json) {
    return json == null ? new List<TagModify>() : json.map((value) => new TagModify.fromJson(value)).toList();
  }

  static Map<String, TagModify> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, TagModify>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new TagModify.fromJson(value));
    }
    return map;
  }
}

